src/bundles/peer-locations.js

```diff
@@ -114,6 +124,44 @@ const parseLatency = (latency) => {
   return value
 }
 
+function getPublicIP (identity) {
+  if (!identity) return
+
+  for (const maddr of identity.addresses) {
+    try {
+      const addr = Multiaddr(maddr).nodeAddress()
+      if (!ip.isPrivate(addr.address)) {
+        return addr.address
+      }
+    } catch (_) {}
+  }
+}
+
+function isPrivateAndNearby (maddr, myIP) {
+  let isPrivate = null
+  let isNearby = null
+
+  try {
+    const addr = maddr.nodeAddress()
+    isPrivate = ip.isPrivate(addr.address)
+
+    if (myIP) {
+      if (addr.family === 4) {
+        isNearby = ip.cidrSubnet(`${myIP}/24`).contains(addr.address)
+      } else if (addr.family === 6) {
+        isNearby = ip.cidrSubnet(`${myIP}/48`).contains(addr.address) &&
+          !ip.cidrSubnet('fc00::/8').contains(addr.address)
+        // peerIP6 ∉ fc00::/8 to fix case of cjdns where IPs are not spatial allocated.
+      }
+    }
+  } catch (_) {
```

Comment:
What exactly might throw here? And are the exceptions related to isPrivate/isNearby?

The try block makes it difficult to see whether or not there's a possibility for a null value to be returned (assuming that's not ok).

I recommend narrowing the try block(s) around specific exceptions and documenting what they are.

Maybe also initialize the values to false to guarantee a safe/usable return value (seems safer, from quick glance), instead of as a side-effect of maybe-related errors.

Reply:
I just updated the function. Please take a look now :D

Reply:
Much clearer, thank you!